docs/docs/reference/metaprogramming/inline.md

```diff
@@ -121,23 +121,29 @@ power(expr, 10)
 ```
 
 Parameters of inline methods can have an `inline` modifier as well. This means
-that actual arguments to these parameters will be inlined in the body of the `inline def`.
-`inline` parameter have call semantics equivalent to by-name parameters but allows for duplication
-of the code in the argument. It is usualy useful constant values need to be propagated to allow
-further optimizations/reductions.
+that actual arguments to these parameters will be inlined in the body of the 
+`inline def`. `inline` parameters have call semantics equivalent to by-name parameters 
+but allow for duplication of the code in the argument. It is usualy useful constant 
+values need to be propagated to allow further optimizations/reductions.
 
 The following example shows the difference in translation between by-value, by-name and `inline`
 parameters:
 
 ```scala
-inline def sumTwice(a: Int, b: =>Int, inline c: Int) = a + a + b + b + c + c
+inline def specialSum(a: Int, b: =>Int, inline c: Int) =
```

Comment:
We can do something even more meaningful. I propose the following snippets:
```scala
inline def funkyAssertEquals(actual: Double, expected: =>Double, inline delta: Double): Unit =
  if (actual - expected).abs > delta then
    throw new AssertionError(s"difference between ${expected} and ${actual} was larger than ${delta}")
```

```scala
funkyAssertEquals(computeActual(), computeExpected(), computeDelta())
// translates to
//
//    val actual = computeActual()
//    def expected = computeExpected()
//    if (actual - expected).abs > computeDelta() then
//      throw new AssertionError(s"difference between ${expected} and ${actual} was larger than ${computeDelta()}")
```

Note: by-name parameters do get a `def`